eth/eventhandler/handlers.go

```diff
@@ -52,15 +53,28 @@ func (eh *EventHandler) handleOperatorAdded(txn basedb.Txn, event *contract.Cont
 		ID:           event.OperatorId,
 	}
 
-	// throw an error if there is an existing operator with the same public key and different operator id
-	operatorData := eh.operatorDataStore.GetOperatorData()
-	if operatorData.ID != 0 && bytes.Equal(operatorData.PublicKey, event.PublicKey) && operatorData.ID != event.OperatorId {
+	// throw an error if operator with the same operator id already exists
+	existsById, err := eh.nodeStorage.OperatorsExist(txn, []spectypes.OperatorID{event.OperatorId})
+	if err != nil {
+		return fmt.Errorf("could not check if operator exists: %w", err)
+	}
+	if existsById {
+		logger.Warn("malformed event: operator registered with ID",
+			zap.Uint64("expected_operator_id", event.OperatorId))
+		return &MalformedEventError{Err: ErrAlreadyRegisteredWithSameId}
+	}
+
+	// throw an error if there is an existing operator with the same public key
+	operatorData, foundByPubKey, err := eh.nodeStorage.GetOperatorDataByPubKey(txn, event.PublicKey)
+	if err != nil {
+		return fmt.Errorf("could not get operator data by public key: %w", err)
+	}
+	if foundByPubKey && operatorData.ID != 0 && bytes.Equal(operatorData.PublicKey, event.PublicKey) && operatorData.ID != event.OperatorId {
 		logger.Warn("malformed event: operator registered with the same operator public key",
 			zap.Uint64("expected_operator_id", operatorData.ID))
-		return &MalformedEventError{Err: ErrAlreadyRegistered}
+		return &MalformedEventError{Err: ErrAlreadyRegisteredWithSamePubkey}
```

Comment:
if the contract does not prevent that, should we? @y0sher @GalRogozinski 

btw, if we choose to not prevent that, we should check around in the entire codebase that we never identify operators by their pubkey (via lookup or by maintaining a map of operators by pubkey, for example)

Reply:
also what would that mean for nodes which synced before this change comes up?

if there's already an active operator with a duplicate pubkey by the time this releases, then newly synced nodes will not see them while already synced nodes would

Reply:
never mind

@mtabasco approves that the contract doesn't allow duplicate registrations so we can proceed :-)

Reply:
> if the contract does not prevent that, should we? @y0sher @GalRogozinski
> 
> btw, if we choose to keep allowing duplicates, we should check around in the entire codebase that we always identify operators by their ID and never by their pubkey (i.e never lookup by pubkey or maintain a map of operators by pubkey, for example)

its like nonce, enforced by the node. 




> also what would that mean for nodes who synced before this change?
> 
> if there's already an active operator with a duplicate pubkey by the time this releases, then newly synced nodes will not see them while already synced nodes would, and it becomes a fork?

I agree about this. but this is a bug, we should check the db before releasing to see if there's any special treatment needed.

Reply:
> I agree about this. but this is a bug, we should check the db before releasing to see if there's any special treatment needed.

@y0sher, let's create a issue for this?

Reply:
@y0sher there's no need to support that now that Marco approved it can't happen bc the contract enforces

Reply:
@moshe-blox so this means we can remove this check as whole? so no need for this PR. (we still need a PR to remove it..)

Reply:
@y0sher technically if we're certain the contract should not emit such events then we should actually return a normal error to fail the sync entirely (rather than malformedError) in these cases

Reply:
@y0sher technically if we're certain the contract should not emit such events then we should actually return a normal error to fail the sync entirely (rather than malformedError) in these cases